numpy/lib/tests/test_function_base.py

```diff
@@ -3095,6 +3095,13 @@ def test_no_p_overwrite(self):
         np.quantile(np.arange(100.), p, interpolation="midpoint")
         assert_array_equal(p, p0)
 
+    def test_quantile_monotic(self):
```

Comment:
I actually thought of writing this as a hypothesis test but hypothesis module was never imported and I don't know your conventions. But can change it hypothesis for sure if that is desired.

Reply:
We recently decided we were willing to accept hypothesis tests - if you think the test is easier to read with hypothesis, please use it, so that maintainers like me who haven't used it can get the hang of it.

Reply:
I added a hypothesis test. Am curious for your feedback. The purpose of this test is not to increase readability but to cover more test cases without writing multiple tests and to find failing egde cases we might not have thought of.